### PR TITLE
[v5.1] Backports for v5.1.2

### PR DESCRIPTION
**Pull request by mheon**
All the stuff from Bug Week that backported cleanly, plus a c/image bump at request of @mtrmac 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -814,7 +814,7 @@ podman_machine_mac_task:
     clone_script:  # artifacts from osx_alt_build_task
         - mkdir -p $CIRRUS_WORKING_DIR
         - cd $CIRRUS_WORKING_DIR
-        - $ARTCURL/OSX%20Cross/repo/repo.tbz
+        - $ARTCURL/Build%20for%20MacOS%20amd64%2Barm64/repo/repo.tbz
```
Comment (Luap99) [inline .cirrus.yml:817]: This will not be correct unless you also backport the task rename but this is overkill so just drop that here

**Review by Luap99 — COMMENTED** [review]
I think you need https://github.com/containers/podman/pull/23013 to fix validate ci error.

Also while you are at it might want to take https://github.com/containers/podman/pull/23066 as well

**Comment by Luap99** [comment]
Ah https://github.com/containers/podman/pull/23017 might be needed as well to get the WSL test passing

File: .cirrus.yml (hunk 1 above)
Comment (Luap99) [reply .cirrus.yml]: I am mean you could have just dropped https://github.com/containers/podman/pull/23240/commits/831baea60e97efe1cb051ccdec02b748786d8bf7 but I well manually reverting works too.

**Comment by Luap99** [comment]
You can rebase now which should hopefully then have all CI problems fixed.

Is there are reason you did not add the version bump (tag) commit into this PR?

**Comment by mheon** [comment]
Wanted to merge this morning, not yesterday. I'm respinning now and including the version bump.

**Comment by Luap99** [comment]
Ah sorry I just looked at the epel failures and that reminded me of https://github.com/containers/podman/pull/23116/commits/6db8ff7f7b1375619fc8f4aed90e5aa220c67a6b

I am not sure why our fedora builds work but without this patch some distros may not be able to build podman.

**Comment by mheon** [comment]
Added

**Review by Luap99 — COMMENTED** [review]
LGTM

**Pull request merged**